system/jlib/jlog.hpp

```diff
@@ -796,23 +806,23 @@ extern jlib_decl ILogMsgHandler * attachLogMsgMonitorFromPTree(IPropertyTree * t
 extern jlib_decl void attachManyLogMsgMonitorsFromPTree(IPropertyTree * tree);            // Takes tree containing many <monitor> elements
 
 // Standard categories and unknown jobInfo
-constexpr LogMsgCategory MCdisaster(MSGAUD_all, MSGCLS_disaster);
-constexpr LogMsgCategory MCuserError(MSGAUD_user, MSGCLS_error);
-constexpr LogMsgCategory MCoperatorError(MSGAUD_operator, MSGCLS_error);
-constexpr LogMsgCategory MCinternalError(MSGAUD_programmer, MSGCLS_error, 1);
-constexpr LogMsgCategory MCauditError(MSGAUD_audit, MSGCLS_error);
-constexpr LogMsgCategory MCuserWarning(MSGAUD_user, MSGCLS_warning);
-constexpr LogMsgCategory MCoperatorWarning(MSGAUD_operator, MSGCLS_warning);
-constexpr LogMsgCategory MCinternalWarning(MSGAUD_programmer, MSGCLS_warning, 1);
-constexpr LogMsgCategory MCauditWarning(MSGAUD_audit, MSGCLS_warning);
-constexpr LogMsgCategory MCuserProgress(MSGAUD_user, MSGCLS_progress);
-constexpr LogMsgCategory MCoperatorProgress(MSGAUD_operator, MSGCLS_progress);
-constexpr LogMsgCategory MCdebugProgress(MSGAUD_programmer, MSGCLS_progress);
-constexpr LogMsgCategory MCuserInfo(MSGAUD_user, MSGCLS_information);
-constexpr LogMsgCategory MCdebugInfo(MSGAUD_programmer, MSGCLS_information);
-constexpr LogMsgCategory MCauditInfo(MSGAUD_audit, MSGCLS_information);
-constexpr LogMsgCategory MCstats(MSGAUD_operator, MSGCLS_progress);
-constexpr LogMsgCategory MCoperatorInfo(MSGAUD_operator, MSGCLS_information);
+constexpr LogMsgCategory MCdisaster(MSGAUD_all, MSGCLS_disaster, FatalMsgThreshold);
+constexpr LogMsgCategory MCuserError(MSGAUD_user, MSGCLS_error, ErrMsgThreshold);
+constexpr LogMsgCategory MCoperatorError(MSGAUD_operator, MSGCLS_error, ErrMsgThreshold);
+constexpr LogMsgCategory MCinternalError(MSGAUD_programmer, MSGCLS_error, AllMsgThreshold);
+constexpr LogMsgCategory MCauditError(MSGAUD_audit, MSGCLS_error, ErrMsgThreshold-10);
```

Comment:
A bit confusing. To this sets it to 9, so a lower threshold than MCuserError. 
I don't you gain any clarity by separating out the level of audit vs user levels, which can already be separated out by audience.

Reply:
I agree here as well. I think the category itself is rather confusing as it is - I think it means it is a high priority audit level event....  maybe this belongs to the AllMsgThreshold.  or maybe just above that threshold AllMsgThreshold+10

Reply:
not sure, audit may or may not be wanted, vs diagnostic logging for example.
Audit logging historically has gone to Dali, and written to a separate audit log trail.
You could argue that the difference audiences, should be able to have different logging levels, so that you could turn off user, or operator logging for example.

Reply:
You might be 100% correct, It might not be wanted but this change isn't intended to edit the predefined logmessage categories. I'm only assigning levels in order to make the logging.details operable. 

With that said, I did take a peek and that particular category isn't used much, I only found one instance of it:
in jthread: AERRLOG("Unauthorized pipe program(%s)",head.str());

That seems to imply a different type of audit. 
For this issue I'm assigning that log message category to  AllMsgThreshold.

Reply:
It seems our definition of "internal" means "an internal programming warning/error" and not "serious internal error"
The vast majority of messages in this category should be in a lower priority than FMCatal* 

I'm not sure if this breaks from convention others use in general, or if it simply depends on different definition of "internal" and there is no concise formal meaning.

To keep things consistent, we should decide in this PR, whether to change all our uses of MCInternal, if they are misleading/against best practices, to a lower priority category.
Or, change the MCInternal threshold from being same as FatalMsgThreshold as it is as the moment, to a threshold level that better fits the majority of it's use cases, e.g. ProgressMsgThreshold.

The latter is the simplest change right now, and since the default maxDetail means it will not suppress any logging anyway, it will only make any difference is logging detail is turned down.
We can then think about revisiting and changing the meaning of how MCInternal is used if we want later in a separate JIRA/PR.